zipline/protocol.py

```diff
@@ -169,6 +169,21 @@ def __repr__(self):
         },
     )
 
+    @property
+    def current_portfolio_weights(self):
+        def asset_multiplier(asset):
+            return asset.multiplier if isinstance(asset, Future) else 1
```

Comment:
Would it be worth pulling this out of the property, so we don't have to redefine the function on each access?